packages/react-server/core/ClientController.js

```diff
@@ -391,31 +391,63 @@ class ClientController extends EventEmitter {
 
 			// During client transitions we create our root
 			// elements as we go.
-			if (!root && this._previouslyRendered){
-				if (element.containerOpen){
-
-					// If we're opening a container that's
-					// our new mountNode.
-					mountNode = this._createContainerNode(
-						mountNode,
-						element.containerOpen,
-						index
+			if (!root && this._previouslyRendered) {
+
+				// If the _previous_ render had elements that
+				// we can re-use we'll render into them.
+				//
+				// DOM re-use is currently opt-in.
+				//
+				if (this._reuseDom) {
+					var oldRootElement = document.querySelector(
+						`div[${REACT_SERVER_DATA_ATTRIBUTE}="${index}"]`
+					);
+					var oldRootContainer = document.querySelector(
+						`div[${PAGE_CONTAINER_NODE_ID}="${index}"]`
```

Comment:
Code style nit: defining these `vars` inside the if-statement with intent to use them outside. Works in JS, but is not what I would call expected. :)